hw06-solid/src/test/java/ru/otus/hw06/api/AtmTest.java

```diff
@@ -0,0 +1,96 @@
+package ru.otus.hw06.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import ru.otus.hw06.api.core.atm.CashCellAtmFactory;
+import ru.otus.hw06.api.core.cell.CashCellFactory;
+import ru.otus.hw06.api.core.money.CashFactory;
+
+class AtmTest {
+
+    // Объект класса АТМ 
+    private Atm atm;
+    @BeforeEach
+    void setupAnEmptyAtm(){
+        List<Cell> cells = new ArrayList<>();
+        cells.add( CashCellFactory.createInfiniteWithMoney( CashFactory.createNoneWithNominal(  50 ) ) );
+        cells.add( CashCellFactory.createInfiniteWithMoney( CashFactory.createNoneWithNominal( 100 ) ) );
+        cells.add( CashCellFactory.createInfiniteWithMoney( CashFactory.createNoneWithNominal( 200 ) ) );
+        cells.add( CashCellFactory.createInfiniteWithMoney( CashFactory.createNoneWithNominal( 500 ) ) );
+        cells.add( CashCellFactory.createInfiniteWithMoney( CashFactory.createNoneWithNominal( 1000 ) ) );
+        cells.add( CashCellFactory.createInfiniteWithMoney( CashFactory.createNoneWithNominal( 2000 ) ) );
+        atm = CashCellAtmFactory.createWithCells( cells );
+    }
+
+    @DisplayName("Принять банкноты разных номиналов")
+    @Test
+    void takeDifferentNominals() {
+        double expectedValue = 50 + 100 + 200 + 500 + 1000 + 2000;
+        
+        Money cash = CashFactory.createSingleWithNominal(  50 );
+        atm.addMoney( cash );
+        cash.setQuantity(100);
+
+        atm.addMoney( CashFactory.createSingleWithNominal( 100 ) );
+        atm.addMoney( CashFactory.createSingleWithNominal( 200 ) );
+        atm.addMoney( CashFactory.createSingleWithNominal( 500 ) );
+        atm.addMoney( CashFactory.createSingleWithNominal( 1000 ) );
+        atm.addMoney( CashFactory.createSingleWithNominal( 2000 ) );
+
+        assertThat( expectedValue ).isEqualTo( atm.getAmount() );
+    }
+
+    @DisplayName("Выдать запрошенную сумму минимальным количеством банкнот")
+    @Test
+    void giveRequestedAmount() {
+        double initialCount = 300 + 60 + 12 + 3 + 2;
+        double expectedValue = initialCount - 2 - 3 - 3 - 1;
```

Comment:
I detect that this code is problematic. According to the [Correctness (CORRECTNESS)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#correctness-correctness), [SA: Nonsensical self computation involving a variable (e.g., x & x) (SA_LOCAL_SELF_COMPUTATION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#sa-nonsensical-self-computation-involving-a-variable-e-g-x-x-sa-local-self-computation).
This method performs a nonsensical computation of a local variable with another reference to the same variable (e.g., x&x or x-x). Because of the nature of the computation, this operation doesn't seem to make sense, and may indicate a typo or a logic error. Double check the computation.